python/ray/node.py

```diff
@@ -102,6 +103,23 @@ def raylet_socket_name(self):
         """Get the node's raylet socket name."""
         return self._raylet_socket_name
 
+    def prepare_socket_file(self, socket_path):
+        """Prepare the socket file for raylet and plasma.
+
+        This method helps to prepare a socket file.
+        1. Make the directory if the directory does not eixt.
+        2. Remove the socket file if the file exists.
+
+        Args:
+            socket_path (string): the socket file to prepare.
+        """
+        if not os.path.isfile(socket_path):
+            path = os.path.dirname(socket_path)
+            if not os.path.isdir(path):
+                try_to_create_directory(path)
+        else:
+            os.remove(socket_path)
```

Comment:
I'm not sure if it's a good idea to remove the existing socket file. If a raylet is still running and user starts ray again, this will break the previous raylet. 
We can check if this socket file is still used by any processes before removing it. Or maybe simply raise an error if it already exists.

Reply:
Because this function is used in raylet socket and plasma socket, it not easy to connect to it to check whether it is in use. I raise exception here. It is better than nothing. If we don't do this and start cluster using `ray start`, there will be no error after this command. We need to check it in raylet.err. With this exception, `ray start` will stop immediately.

Reply:
Yep, raising the error earlier is better. 
BTW, we can detect if a socket file is in use with the `lsof` command. But raising an error is okay as well.